specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2023-03-01/checkPolicyRestrictions.json

```diff
@@ -0,0 +1,542 @@
+        },
+        "policy": {
+          "description": "The details of the policy that is causing the field restriction.",
+          "$ref": "#/definitions/PolicyReference",
+          "readOnly": true
+        },
+        "policyEffect": {
+          "description": "The effect of the policy that is causing the field restriction.",
+          "$ref": "#/definitions/PolicyEffect",
+          "readOnly": true
+        },
+        "reason": {
+          "description": "The reason for the restriction.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "PolicyEvaluationResult": {
+      "description": "The result of a non-compliant policy evaluation against the given resource content.",
+      "properties": {
+        "policyInfo": {
+          "description": "The details of the policy that was evaluated.",
+          "$ref": "#/definitions/PolicyReference",
+          "readOnly": true
+        },
+        "evaluationResult": {
+          "description": "The result of the policy evaluation against the resource. This will typically be 'NonCompliant' but may contain other values if errors were encountered.",
+          "type": "string",
+          "readOnly": true
+        },
+        "evaluationDetails": {
+          "description": "The detailed results of the policy expressions and values that were evaluated.",
+          "$ref": "#/definitions/PolicyEvaluationDetails",
+          "readOnly": true
+        },
+        "effectDetails": {
+          "description": "The details of the effect that was applied to the resource.",
+          "$ref": "#/definitions/PolicyEffectDetails",
+          "readOnly": true
+        }
+      }
+    },
+    "PolicyReference": {
+      "description": "Resource identifiers for a policy.",
+      "properties": {
+        "policyDefinitionId": {
+          "description": "The resource identifier of the policy definition.",
+          "type": "string",
+          "readOnly": true
+        },
+        "policySetDefinitionId": {
+          "description": "The resource identifier of the policy set definition.",
+          "type": "string",
+          "readOnly": true
+        },
+        "policyDefinitionReferenceId": {
+          "description": "The reference identifier of a specific policy definition within a policy set definition.",
+          "type": "string",
+          "readOnly": true
+        },
+        "policyAssignmentId": {
+          "description": "The resource identifier of the policy assignment.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "PolicyEffectDetails": {
+      "type": "object",
+      "description": "The details of the effect that was applied to the resource.",
+      "properties": {
+        "policyEffect": {
+          "description": "The effect that was applied to the resource",
+          "$ref": "#/definitions/PolicyEffect",
+          "readOnly": true
+        }
+      }
+    },
+    "PolicyEffect": {
```

Comment:
if this enum is in swagger any new effects will require a rev to the API version of this API and should be excluded from older API versions.  I don't know if this is worth the benefit of having it as an enum.  

Effect is a fairly dynamic field that isn't directly tied to this API.  Its more tied to entities related to what it reports on (i.e. the policies themselves).

Reply:
In that case, would you recommend a simply using `string` for `policyEffect`?

Reply:
yeah.  Maybe put a link to the "policy effect" docs in the description of the property